### PR TITLE
[datafeeder]: Local storage

### PR DESCRIPTION
**Pull request by y-vynohradov**
refactor: Change local storage structure

### CODE REVIEW

File: libs/editor/src/lib/services/wizard.service.ts [hunk 1]
```diff
@@ -13,8 +13,25 @@ export class WizardService {
...
-    this.initialize()
+  private id: string
+
+  constructor(private translateService: TranslateService) {}
+
+  initialize(id: string): void {
+    this.id = id
+    this.wizardData.clear()
+
+    const savedWizardData = localStorage.getItem(id)
```
Comment (fgravin) [inline libs/editor/src/lib/services/wizard.service.ts:24]: almost, the id of the batch should not be ids of the localStorage, but ids of the dictionnary stored in the localStorage
```js
localStorage
 - datafeeder-state: {
     d7c14181-2e66-49ff-868e-8c55671f955f: [{},{}]...,     
     trtrez543534543543534: [{},{}]...,
}
```

File: libs/editor/src/lib/services/wizard.service.ts [hunk 2]
```diff
@@ -39,21 +56,35 @@ export class WizardService {
...
+    const config = []
+
+    this.wizardData.forEach((value, key) => {
+      config.push({
+        id: key,
+        value,
+      })
+    })
+
+    const data = {
```
Comment (fgravin) [inline libs/editor/src/lib/services/wizard.service.ts:76]: why do you save the step exactly ? how is using it ?

**Review by fgravin — CHANGES_REQUESTED** [review]
Quite good but not exactly the good structure still.
Please add some test as well.

Also I don't really understand the purpose to store the step in the localStorage as it is already in the url, explain me you approach please.

Thanks

**Comment by fgravin** [comment]
Also please rebase on master, thanks

File: libs/editor/src/lib/components/wizard-summarize/wizard-summarize.component.spec.ts [hunk 3]
```diff
@@ -8,14 +8,38 @@ import { HttpClientTestingModule } from '@angular/common/http/testing'
...
 
-const localStorageMock = () => {
-  let storage = {}
-  return {
-    getItem: (key) => (key in storage ? storage[key] : null),
-    setItem: (key, value) => (storage[key] = value || ''),
-    removeItem: (key) => delete storage[key],
-    clear: () => (storage = {}),
-  }
+const wizardServiceMock = {
```
Comment (fgravin) [inline libs/editor/src/lib/components/wizard-summarize/wizard-summarize.component.spec.ts:11]: good !
but you should test that it is called with the correct params.

File: libs/editor/src/lib/services/wizard.service.ts [hunk 4]
```diff
@@ -13,8 +13,25 @@ export class WizardService {
   private wizardStep = 1
   private wizardData: Map<string, any> = new Map()
 
-  constructor(private translateService: TranslateService) {
-    this.initialize()
+  private id: string
+
+  constructor(private translateService: TranslateService) {}
```
Comment (fgravin) [inline libs/editor/src/lib/services/wizard.service.ts:18]: please add tests in this service, where you mock the localStorage

**Review by fgravin — CHANGES_REQUESTED** [review]
Thanks for the refactoring.
You should add some tests please.

**Comment by tonio** [comment]
Integrated into #103

**Pull request closed without merging**